crates/autopilot/src/decoded_settlement.rs

```diff
@@ -40,16 +40,20 @@ type DecodedSettlementTokenized = (
     [Vec<(Address, U256, Bytes<Vec<u8>>)>; 3],
 );
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Eq)]
 pub struct DecodedSettlement {
     // TODO check if `EncodedSettlement` can be reused
     pub tokens: Vec<Address>,
     pub clearing_prices: Vec<U256>,
     pub trades: Vec<DecodedTrade>,
     pub interactions: [Vec<DecodedInteraction>; 3],
+    /// Data that was appended to the regular call data of the `settle()` call
+    /// as a form of on-chain meta data. This gets used to associated a
+    /// settlement with an auction.
+    pub metadata: Option<Bytes<[u8; Self::META_DATA_LEN]>>,
```

Comment:
I don't like that we're inventing this new concept of "settlement meta data". The last few bytes are just the auction ID. This is known ahead of time.

Reply:
Yeah, this felt more correct when I tried to implement a more generic solution. But because that had subtle issues I went for the bare bones where it indeed makes more sense to simply call that settlement id.